handlers.go

```diff
@@ -201,94 +192,6 @@ func setupGatewayHandler(cfg Config, nd *Node) (http.Handler, error) {
-	// are transformed to GET requests to the gateway.
-	// - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/303
-	redirectToGateway := func(format string) http.HandlerFunc {
-		return func(w http.ResponseWriter, r *http.Request) {
-			path, err := path.NewPath(prefixPath(r.URL.Query().Get("arg")))
-			if err != nil {
-				w.WriteHeader(http.StatusBadRequest)
-				w.Write([]byte(err.Error()))
-				return
-			}
-			url := path.String()
-			if format != "" {
-				url += "?format=" + format
-			}
-
-			goLog.Debugw("api request redirected to gateway", "url", r.URL, "redirect", url)
-			http.Redirect(w, r, url, http.StatusSeeOther)
-		}
-	}
-
-	mux.HandleFunc("/api/v0/cat", redirectToGateway(""))
-	mux.HandleFunc("/api/v0/dag/export", redirectToGateway("car"))
-	mux.HandleFunc("/api/v0/block/get", redirectToGateway("raw"))
-	mux.HandleFunc("/api/v0/dag/get", func(w http.ResponseWriter, r *http.Request) {
-		path, err := path.NewPath(prefixPath(r.URL.Query().Get("arg")))
-		if err != nil {
-			w.WriteHeader(http.StatusBadRequest)
-			w.Write([]byte(err.Error()))
-			return
-		}
-		codec := r.URL.Query().Get("output-codec")
-		if codec == "" {
-			codec = "dag-json"
-		}
-		url := fmt.Sprintf("%s?format=%s", path.String(), codec)
-		goLog.Debugw("api request redirected to gateway", "url", r.URL, "redirect", url)
-		http.Redirect(w, r, url, http.StatusSeeOther)
-	})
-
-	// Endpoints that have high traffic volume. We will keep redirecting these
-	// for now to Kubo endpoints that are able to handle these requests. We use
-	// 307 Temporary Redirect in order to preserve the original HTTP Method.
-	// - https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307
-	s := rand.NewSource(time.Now().Unix())
-	rand := rand.New(s)
-	redirectToKubo := func(w http.ResponseWriter, r *http.Request) {
-		// Naively choose one of the Kubo RPC clients.
-		endpoint := endpoints[rand.Intn(len(endpoints))]
-		url := endpoint + r.URL.Path
-		if r.URL.RawQuery != "" {
-			url += "?" + r.URL.RawQuery
-		}
-		goLog.Debugw("api request redirected to kubo", "url", r.URL, "redirect", url)
-		http.Redirect(w, r, url, http.StatusTemporaryRedirect)
-	}
-
-	mux.HandleFunc("/api/v0/name/resolve", redirectToKubo)
-	mux.HandleFunc("/api/v0/name/resolve/", redirectToKubo)
-	mux.HandleFunc("/api/v0/resolve", redirectToKubo)
-	mux.HandleFunc("/api/v0/dag/resolve", redirectToKubo)
-	mux.HandleFunc("/api/v0/dns", redirectToKubo)
-
-	// Remaining requests to the API receive a 501, as well as an explanation.
-	mux.HandleFunc("/api/v0/", func(w http.ResponseWriter, r *http.Request) {
-		w.WriteHeader(http.StatusNotImplemented)
-		goLog.Debugw("api request returned 501", "url", r.URL)
-		w.Write([]byte("The /api/v0 Kubo RPC is now discontinued on this server as it is not part of the gateway specification. If you need this API, please self-host a Kubo instance yourself: https://docs.ipfs.tech/install/command-line/"))
-	})
```

Comment:
Can we keep this message in a static handler for  all responses returned for `/api/v0/*`? 

> `The /api/v0 Kubo RPC is not part of IPFS Gateway. If you need this API, please self-host a Kubo instance yourself: https://docs.ipfs.tech/install/command-line/ with proper auth https://github.com/ipfs/kubo/blob/master/docs/config.md#apiauthorizations`